source/Ship.cpp

```diff
@@ -2728,15 +2728,16 @@ bool Ship::FireAntiMissile(const Projectile &projectile, vector<Visual> &visuals
 	if(CannotAct())
 		return false;
 
-	double jamChance = CalculateJamChance(Energy(), ionization);
-
-	const vector<Hardpoint> &hardpoints = armament.Get();
-	for(unsigned i = 0; i < hardpoints.size(); ++i)
+	for(auto *hardpoint : armament.AntiMissileWeapons())
 	{
-		const Weapon *weapon = hardpoints[i].GetOutfit();
-		if(weapon && CanFire(weapon))
+		const Weapon *weapon = hardpoint->GetOutfit();
+		if(CanFire(weapon))
+		{
+			const double jamChance = CalculateJamChance(Energy(), ionization);
+			int i = armament.WeaponIndex(*hardpoint);
 			if(armament.FireAntiMissile(i, *this, projectile, visuals, Random::Real() < jamChance))
```

Comment:
Why pass the index and not the hardpoint directly and use that? Same goes for the Fire function as well.